tf_projection_rostest.test outputs huge amount of warning with tf2 0.7.1

https://github.com/at-wat/neonavigation/pull/493

tf2 (geometry2) 0.7.1 update causes it and the CI job fails due to too large log.
```
rostest log file is in /home/buildfarm/.ros/log/rostest-d0db83d9ae99-13317.log
-- run_tests.py: verify result "/tmp/ws/test_results/track_odometry/rostest-test_test_track_odometry_rostest.xml"
Built target _run_tests_track_odometry_rostest_test_test_track_odometry_rostest.test
Scanning dependencies of target _run_tests_track_odometry_rostest_test_test_tf_projection_rostest.test
-- run_tests.py: execute commands
  /usr/bin/python3 /opt/ros/noetic/share/rostest/cmake/../../../bin/rostest --pkgdir=/tmp/ws/src/neonavigation__custom-2/track_odometry --package=track_odometry --results-filename test_test_tf_projection_rostest.xml --results-base-dir "/tmp/ws/test_results" /tmp/ws/src/neonavigation__custom-2/track_odometry/test/test/tf_projection_rostest.test 
... logging to /home/buildfarm/.ros/log/rostest-d0db83d9ae99-13424.log
[ROSUNIT] Outputting test results to /tmp/ws/test_results/track_odometry/rostest-test_test_tf_projection_rostest.xml
[31m[ERROR] [1589498236.476909854]: tf_projection parameters "base_link_frame", "projection_frame", "target_frame", and "frame" are replaced by "source_frame", "projection_surface_frame", "parent_frame", and "projected_frame"[0m
[33m[ WARN] [1589498236.709440848]: "map" passed to lookupTransform argument target_frame does not exist. [0m
[33m[ WARN] [1589498236.742234377]: "map" passed to lookupTransform argument target_frame does not exist. [0m
[33m[ WARN] [1589498236.814686772]: "map" passed to lookupTransform argument target_frame does not exist. [0m
[33m[ WARN] [1589498236.817170210]: "map" passed to lookupTransform argument target_frame does not exist. [0m
Warning: Warning: Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected at time 0.100000 according to authority unknown_publisherTF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected at time 0.100000 according to authority unknown_publisher
Warning: 
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected2 at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected2 at time 0.100000 according to authority unknown_publisher
         at line 278 in Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected2 at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
/tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_projected2 at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_tilt_projected at time 0.100000 according to authority unknown_publisher
         at line Warning: 278TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_tilt_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cppWarning: 
TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_tilt_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_tilt_projected at time 0.100000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_tilt_projected_with_aligned at time 0.100000 according to authority unknown_publisherTF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link_tilt_projected_with_aligned at time 0.100000 according to authority unknown_publisher

         at line          at line 278278 in  in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp/tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
```
...
```
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link at time 1589498429.384536 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link at time 1589498429.394536 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.1/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_link at time 1589498429.404529 accord

The job exceeded the maximum log length, and has been terminated.
```

--- a/planner_cspace/include/planner_cspace/jump_detector.h
+++ b/planner_cspace/include/planner_cspace/jump_detector.h
@@ -47,7 +47,6 @@ class JumpDetector
 {
 protected:
   tf2_ros::Buffer& tfbuf_;
-  tf2_ros::TransformListener tfl_;
   tf2::Transform base_trans_prev_;
   tf2::Transform map_trans_prev_;
 
@@ -62,7 +61,6 @@ protected:
 public:
   explicit JumpDetector(tf2_ros::Buffer& tfbuf)
     : tfbuf_(tfbuf)
-    , tfl_(tfbuf_)
     , base_trans_prev_(tf2::Quaternion(0, 0, 0, 1))
     , map_trans_prev_(tf2::Quaternion(0, 0, 0, 1))
     , init_(false)

--- a/planner_cspace/test/src/test_navigate_boundary.cpp
+++ b/planner_cspace/test/src/test_navigate_boundary.cpp
@@ -64,7 +64,7 @@ protected:
   void publishTransform(const double x, const double y)
   {
     geometry_msgs::TransformStamped trans;
-    trans.header.stamp = ros::Time();
+    trans.header.stamp = ros::Time::now();
     trans.header.frame_id = "odom";
     trans.child_frame_id = "base_link";
     trans.transform.translation.x = x;
@@ -120,7 +120,7 @@ TEST_F(NavigateBoundary, StartPositionScan)
 int main(int argc, char** argv)
 {
   testing::InitGoogleTest(&argc, argv);
-  ros::init(argc, argv, "test_navigate");
+  ros::init(argc, argv, "test_navigate_boundary");
 
   return RUN_ALL_TESTS();
 }

--- a/track_odometry/src/tf_projection_node.cpp
+++ b/track_odometry/src/tf_projection_node.cpp
@@ -31,6 +31,7 @@
 
 #include <tf2/utils.h>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <tf2_ros/static_transform_broadcaster.h>
 #include <tf2_ros/transform_broadcaster.h>
 #include <tf2_ros/transform_listener.h>
 
@@ -45,6 +46,7 @@ private:
   ros::NodeHandle pnh_;
   tf2_ros::Buffer tf_buffer_;
   tf2_ros::TransformListener tf_listener_;
+  tf2_ros::StaticTransformBroadcaster tf_static_broadcaster_;
   tf2_ros::TransformBroadcaster tf_broadcaster_;
 
   double rate_;
@@ -108,9 +110,12 @@ public:
     }
     catch (tf2::TransformException& e)
     {
-      ROS_WARN_ONCE("%s", e.what());
+      ROS_WARN_THROTTLE(1.0, "%s", e.what());
       return;
     }
+
+    if (!trans.stamp_.isZero())
+      trans.stamp_ += ros::Duration(tf_tolerance_);
 
     if (project_posture_)
     {
@@ -130,7 +135,7 @@ public:
 
     const tf2::Stamped<tf2::Transform> result(
         track_odometry::projectTranslation(trans, trans_target),
-        trans.stamp_ + ros::Duration(tf_tolerance_),
+        trans.stamp_,
         parent_frame_);
 
     geometry_msgs::TransformStamped trans_out = tf2::toMsg(result);
@@ -141,7 +146,14 @@ public:
     }
     trans_out.child_frame_id = projected_frame_;
 
-    tf_broadcaster_.sendTransform(trans_out);
+    if (trans.stamp_.isZero())
+    {
+      tf_static_broadcaster_.sendTransform(trans_out);
+    }
+    else
+    {
+      tf_broadcaster_.sendTransform(trans_out);
+    }
   }
   void cbTimer(const ros::TimerEvent& event)
   {

--- a/trajectory_tracker/test/src/test_trajectory_tracker.cpp
+++ b/trajectory_tracker/test/src/test_trajectory_tracker.cpp
@@ -59,6 +59,7 @@ private:
   ros::Publisher pub_path_vel_;
   tf2_ros::TransformBroadcaster tfb_;
   ros::Time cmd_vel_time_;
+  ros::Time trans_stamp_last_;
 
 protected:
   std_msgs::Header last_path_header_;
@@ -207,7 +208,12 @@ public:
     trans.transform.rotation.y = q.y();
     trans.transform.rotation.z = q.z();
     trans.transform.rotation.w = q.w();
-    tfb_.sendTransform(trans);
+
+    if (trans.header.stamp != trans_stamp_last_)
+    {
+      tfb_.sendTransform(trans);
+    }
+    trans_stamp_last_ = trans.header.stamp;
   }
 };
 

